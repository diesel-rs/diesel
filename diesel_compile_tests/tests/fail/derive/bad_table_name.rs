#[macro_use]
extern crate diesel;

table! {
    users {
        id -> Integer,
        name -> Text,
    }
}

#[derive(Queryable)]
#[diesel(table_name)]
//~^ ERROR: unexpected end of input, expected `=`
struct User1 {
    name: String,
}

#[derive(Queryable)]
#[diesel(table_name(users))]
//~^ ERROR: expected `=`
struct User2 {
    name: String,
}

#[derive(Insertable)]
#[diesel(table_name = true)]
//~^ ERROR: expected identifier, found keyword `true`
struct User3 {
    id: i32,
}

#[derive(Insertable)]
#[diesel(table_name = "not a path")]
//~^ ERROR:  expected identifier
struct User4 {
    id: i32,
}

#[derive(Insertable)]
#[diesel(table_name = does::not::exist)]
//~^ ERROR: failed to resolve: use of unresolved module or unlinked crate `does`
struct User5 {
    id: i32,
}

fn main() {}
