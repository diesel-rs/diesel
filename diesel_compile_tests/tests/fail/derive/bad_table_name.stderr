error: unexpected end of input, expected `=`
       help: The correct format looks like `#[diesel(table_name = foo)]`
  --> tests/fail/derive/bad_table_name.rs:12:20
   |
12 | #[diesel(table_name)]
   |                    ^

error: expected `=`
  --> tests/fail/derive/bad_table_name.rs:18:20
   |
18 | #[diesel(table_name(users))]
   |                    ^

error: expected identifier, found keyword `true`
  --> tests/fail/derive/bad_table_name.rs:24:23
   |
24 | #[diesel(table_name = true)]
   |                       ^^^^

error: expected identifier
  --> tests/fail/derive/bad_table_name.rs:30:23
   |
30 | #[diesel(table_name = "not a path")]
   |                       ^^^^^^^^^^^^

error[E0433]: failed to resolve: use of undeclared crate or module `does`
  --> tests/fail/derive/bad_table_name.rs:36:23
   |
36 | #[diesel(table_name = does::not::exist)]
   |                       ^^^^ use of undeclared crate or module `does`
